src/components/external/PayByLink/PayByLinkSettings/components/PayByLinkSettingsContainer/context/context.tsx

```diff
@@ -0,0 +1,72 @@
+import { memo, PropsWithChildren } from 'preact/compat';
+import { createContext } from 'preact';
+import { useCallback, useContext, useRef, useState, useEffect } from 'preact/hooks';
+import { noop } from '../../../../../../../utils';
+import { IPayByLinkSettingsContext } from './types';
+import { ActiveMenuItem, DEFAULT_MENU_ITEM } from './constants';
+import { useStores } from '../../../../../../../hooks/useStores';
+
+export const PayByLinkSettingsContext = createContext<IPayByLinkSettingsContext>({
+    activeMenuItem: ActiveMenuItem.theme,
+    payload: undefined,
+    setPayload: noop,
+    selectedStore: undefined,
+    setActiveMenuItem: noop,
+    saveActionCalled: undefined,
+    setIsValid: noop,
+    getIsValid: () => false,
+    setSaveActionCalled: noop,
+    stores: undefined,
+    setSelectedStore: noop,
+});
+
+export const PayByLinkSettingsProvider = memo(({ children }: PropsWithChildren) => {
+    const [activeMenuItem, setActiveMenuItem] = useState<string>(DEFAULT_MENU_ITEM);
+    const [payload, setPayload] = useState(null);
+    const isValid = useRef(false);
+    const [saveActionCalled, setSaveActionCalled] = useState<boolean | undefined>(false);
+    const { stores, selectedStore, setSelectedStore } = useStores();
+
+    console.log('context');
+
+    useEffect(() => {
+        if (!selectedStore) setSelectedStore(stores?.[0]?.id);
+    }, [stores, selectedStore, setSelectedStore]);
+
+    const onPayloadChange = useCallback((payload: any) => {
```

Comment:
```suggestion
    const onPayloadChange = useCallback((payload: PayByLinkSettingsPayload) => {
```

We could make these changes to make the code more type safe.